server/meshmodel/schemas/component.json

```diff
@@ -0,0 +1,57 @@
+{
+  "$id": "https://schema.meshery.io/component.json",
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "description": "Components are the atomic units for designing infrastructure.",
+  "required": [
+    "apiVersion",
+    "kind",
+    "schema",
+    "model"
+  ],
+  "additionalProperties": false,
+  "type": "object",
+  "properties": {
+    "apiVersion": {
+      "type": "string"
+    },
+    "kind": {
+      "type": "string"
+    },
+    "metadata": {
+      "type": "object",
+      "required": [
+        "name",
+        "version"
+      ],
+      "properties": {
+        "description": {
+          "type": "string"
+        },
+        "name": {
```

Comment:
Could you add a regex ```pattern``` where ever it possible .

Reply:
@harkiratsm i had one defined in the first commit. You can grab it and update this PR with your additions of pattern regexes. 

What other suggestions do you have? I already suggested this change.

Reply:
Not much !